### ISSUE TITLE
aria-haspopup value should be set to listbox for BaseExtendedPicker

### ISSUE THREAD

**Issue by syedinaqvi** [microsoft/fluentui#18535]
### Environment Information

- **Package version(s)**: 8.10.0
- **Browser and OS versions**: (fill this out) Edge with Windows Narrator

### Describe the issue:

Bug does repro on Controlled People Picker example here https://developer.microsoft.com/en-us/fluentui#/controls/web/peoplepicker. Setting focus on the combobox field narrator announces name of the field fallowed by combo edit, collapsed has pop-up. but on hitting enter nothing happen. 

### Please provide a reproduction of the issue in a codepen:
https://developer.microsoft.com/en-us/fluentui#/controls/web/peoplepicker. Repro on Controlled People Picker example. 

#### Actual behavior:

Setting focus on the combobox field narrator announces name of the field fallowed by combo edit, collapsed has pop-up. but on hitting enter nothing happen as there is no combobox options available. 

#### Expected behavior:

Has popup should be announced only when popup is available. 

### Documentation describing expected behavior

BaseExtendedPicker has aria-haspopup="true", where as according to https://www.w3.org/TR/wai-aria-1.1/#combobox, "Elements with the role combobox have an implicit aria-haspopup value of listbox.". setting aria-haspopup to a correct value may fix this issue. 
Unified picker set the value for aria-haspopup correct and doesn't have this issue. 
https://github.com/microsoft/fluentui/blob/dab45a2afe609709735b9b6e604a1af40d50e809/packages/react-experiments/src/components/UnifiedPicker/UnifiedPicker.tsx#L15

**Comment by gouttierre** [comment]
@syedinaqvi - Thank you for reaching out to us and filing this issue.

@smhigley - Would you be able to confirm if this is a regression, or if this behavior is an issue?
cc @ling1726

**Comment by smhigley** [comment]
The behavior in the bug description is not the correct expectation for a combobox. All comboboxes announce that they have some type of popup, regardless of the specific mechanism to open it. I agree that not showing the suggestion list until someone starts typing is a bit weird outside of certain specific situations (e.g. an email "to" line), but it's still a combobox with a popup.

Removing `aria-haspopup` isn't going to do anything in any case, since all comboboxes have a default `aria-haspopup` value of `listbox`. You can verify that here, where the combobox does **not** specify `aria-haspopup`, but Narrator still announces that it has a popup (as it should): https://w3c.github.io/aria-practices/examples/combobox/combobox-autocomplete-both.html.

Closing this now.
